src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -145,7 +151,56 @@ private boolean userIsSuperAdmin(User user, AdminDNs adminDNs) {
         return user != null && adminDNs.isAdmin(user);
     }
 
-    private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+    private boolean authorizeRequest(RestHandler original, RestRequest request, RestChannel channel, User user) {
+
+        List<RestHandler.Route> restRoutes = original.routes();
+        Optional<RestHandler.Route> handler = restRoutes.stream()
+            .filter(rh -> rh.getMethod().equals(request.method()))
+            .filter(rh -> restPathMatches(request.path(), rh.getPath()))
+            .findFirst();
+        if (handler.isPresent() && handler.get() instanceof NamedRoute) {
```

Comment:
Invert the flow of control to lower the intending making it easier to read
```suggestion
        if (!handler.isPresent() || !(handler.get() instanceof NamedRoute)) {
           return false;
        }
```

Reply:
Note; you can also use a `final boolean doesRouteSupportRestAuthorization = /* gunk */;` to protect yourself in this case